src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -554,6 +554,17 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
                 hasErrors = true;
             }
 
+            if (node.IsKind(SyntaxKind.EqualsExpression) || node.IsKind(SyntaxKind.NotEqualsExpression))
+            {
+                if (isDoubleNaN(left) || isFloatNaN(left) || isDoubleNaN(right) || isFloatNaN(right))
```

Comment:
Can this apply to `<`, `>`, `<=` and `>=` as well? (See "Comparing with NaN" in https://en.wikipedia.org/wiki/NaN).

Reply:
@jkotas Sure. I'll update it if the warning is approved in future LDM.